contracts/actions/curveusd/helpers/CurveUsdHelper.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.10;
+
+import { ICrvUsdController, ILLAMMA, ICrvUsdControllerFactory, ICurveUsdSwapper } from "../../../interfaces/curveusd/ICurveUsd.sol";
+import "../../../interfaces/curve/IAddressProvider.sol";
+
+import "./MainnetCurveUsdAddresses.sol";
+import "../../../utils/TokenUtils.sol";
+
+contract CurveUsdHelper is MainnetCurveUsdAddresses {
+    using TokenUtils for address;
+
+    error CurveUsdInvalidController();
+
+    IAddressProvider addressProvider = IAddressProvider(CURVE_ADDRESS_PROVIDER);
+
+    bytes4 constant CURVE_SWAPPER_ID = bytes4(keccak256("CurveUsdSwapper"));
+
+    function isControllerValid(address _controllerAddr) public view returns (bool) {
+        return ICrvUsdControllerFactory(CRVUSD_CONTROLLER_FACTORY_ADDR).debt_ceiling(_controllerAddr) != 0;
+    }
+
+    function userMaxWithdraw(address _controllerAddress, address _user) public view returns (uint256 maxWithdraw) {
+        address llammaAddress = ICrvUsdController(_controllerAddress).amm();
```

Comment:
`user_state(user: address) -> uint256[4]` from controller can return all the necessary info for `min_collateral` call